src/Mvc/src/Microsoft.AspNetCore.Mvc.Core/UrlHelperExtensions.cs

```diff
@@ -530,5 +530,57 @@ public static string Page(this IUrlHelper urlHelper, string pageName, object val
                 host: host,
                 fragment: fragment);
         }
+
+        /// <summary>
+        /// Generates an absolute URL for an action method, which contains the specified
+        /// <paramref name="action"/> name, <paramref name="controller"/> name, route <paramref name="values"/>,
+        /// <paramref name="protocol"/> to use, <paramref name="host"/> name, and <paramref name="fragment"/>.
+        /// Generates an absolute URL if the <paramref name="protocol"/> and <paramref name="host"/> are
+        /// non-<c>null</c>. See the remarks section for important security information.
+        /// </summary>
+        /// <param name="helper">The <see cref="IUrlHelper"/>.</param>
+        /// <param name="action">The name of the action method. When not defined, defaults to the current executing action.</param>
+        /// <param name="controller">The name of the controller. When not defined, defaults to the current executing controller.</param>
+        /// <param name="values">An object that contains route values.</param>
+        /// <param name="protocol">The protocol for the URL, such as "http" or "https".</param>
+        /// <param name="host">The host name for the URL.</param>
+        /// <param name="fragment">The fragment for the URL.</param>
+        /// <returns>The generated URL.</returns>
+        /// <remarks>
+        /// <para>
+        /// The value of <paramref name="host"/> should be a trusted value. Relying on the value of the current request
+        /// can allow untrusted input to influence the resulting URI unless the <c>Host</c> header has been validated.
+        /// See the deployment documentation for instructions on how to properly validate the <c>Host</c> header in
+        /// your deployment environment.
+        /// </para>
+        /// </remarks>
+        public static string ActionLink(
+            this IUrlHelper helper,
+            string action = null,
+            string controller = null,
+            object values = null,
+            string protocol = null,
+            string host = null,
+            string fragment = null)
+        {
+            if (helper == null)
+            {
+                throw new ArgumentNullException(nameof(helper));
+            }
+
+            var httpContext = helper.ActionContext.HttpContext;
+
+            if (protocol == null)
```

Comment:
`protocol` and `host` are `string`s. If we go the default value route, use `!string.IsNullOrEmpty(protocol)` and do the same for the `host` check.

Reply:
Technically, passing on `null` or an empty string will cause a different behavior down in `UrlHelperBase` though (not that this would be particularly useful).

Reply:
Not entirely convinced we need to do anything when a user explicitly passes in a value to the method.